lib/connections/util.go

```diff
@@ -32,3 +34,75 @@ func fixupPort(uri *url.URL, defaultPort int) *url.URL {
 
 	return &copyURI
 }
+
+func getURLsForAllAdaptersIfUnspecified(network string, uri *url.URL) []*url.URL {
+	addrs := make([]*url.URL, 0)
+	if ip, port, err := resolve(network, uri.Host); err == nil && port != 0 && (len(ip) == 0 || ip.IsUnspecified()) {
```

Comment:
early return instead of nesting?

Reply:
So just inverting the if got me confused. Aka `!ip.IsUnspecified()` is weird.